### PR TITLE
Support of versions of cp2k below 7 marked as deprecated

### PR DESCRIPTION
**Pull request by mtaillefumier**
Update of the sirius package forces cleaning up the recipe for cp2k.

### CODE REVIEW

**Comment by dev-zero** [comment]
Not really a fan of this, tbh. I was always under the impression that one of the goals of Spack is to maintain access to older versions of a software package.

And, the wannier90 interface goes via files and there is no CP2K preprocessor flag for that, so it is always built-in, AFAIK.

**Comment by toxa81** [comment]
@dev-zero access to old versions is available with tagged releases of spack

**Comment by mtaillefumier** [comment]
Wannier90 : Indeed there is no flags for wannier90. I will remove it

**Comment by dev-zero** [comment]
also, as per policy, those versions first have to be marked with `deprecated=True` first and can then be removed after the next Spack release.

**Comment by akamishra2826** [comment]
With AMD-ZSS-toolchains we are supporting cp2k-7.1 to build and run using spack, why this PR is required please explain in detail ?

**Comment by mtaillefumier** [comment]
I reverted everything and put versions below 9.1 as deprecated

**Comment by mtaillefumier** [comment]
The reason is that one dependency is been cleaned up and that early version of cp2k are still depending on it. I reverted all my changes and just put all versions below 9.1 as deprecated. 

Any reasons why the AMD-ZSS-toolchains need 7.1 ? (NB : cp2k-7.1 is not suppressed, just deprecated)

**Comment by ningli-amd** [comment]
As CP2K uses a lot of modern Fortran features, support to the newer versions by LLVM-based compilers can be problematic. For example, I experienced a number of compile-time and run-time issues with v2023.1. v7.1 can be reliably supported by these compilers. I suggest we keep v7.1 onwards in the Spack recipe until the compilers catch up.

**Comment by mtaillefumier** [comment]
I think I found a compromise. SIRIUS version below 6 are removed from the SIRIUS package (we do not support them anymore) and initial SIRIUS support in CP2K was introduced around version 7 ~ 8 but effectively usable starting from CP2K 9.x. 
I do not think SIRIUS is an issue for compiling CP2K on LLVM based FORTRAN compilers as SIRIUS is pure C++. To be able to update the SIRIUS package and keep changes to CP2K package minimal, SIRIUS 6 dependency is marked as conflict.

**Review by amd-toolchain-support — APPROVED** [review]
This looks fine from our POV. 

In general we would definitely prefer keeping support for packages in Spack for as long as possible. There are definitely cases where running older software versions on the latest architectures is useful and desirable, but we accept that has to be balanced with practicality of maintaining recipes if the complexity starts to grow.

File: var/spack/repos/builtin/packages/cp2k/package.py [hunk 1]
```diff
@@ -231,14 +247,13 @@ class Cp2k(MakefilePackage, CudaPackage, CMakePackage, ROCmPackage):
...
-        depends_on("sirius@:6", when="@:7")
         depends_on("sirius@7.0.0:7.0", when="@8:8.2")
         depends_on("sirius@7.2", when="@8.3:8.9")
         depends_on("sirius@7.3:", when="@9.1")
         depends_on("sirius@7.4:", when="@master")
         conflicts("~mpi", msg="SIRIUS requires MPI")
-        # sirius support was introduced in 7+
-        conflicts("@:6")
+        # sirius support was introduced in 7, but effectively usable starting from CP2K 9
+        conflicts("@:7")
```
Comment (msimberg) [inline var/spack/repos/builtin/packages/cp2k/package.py]:
```suggestion
        conflicts("@:8")
```
? If the intention is to conflict with versions older than 9.

**Review by msimberg — APPROVED** [review]

**Comment by msimberg** [comment]
@dev-zero would you mind having another look?

**Review by dev-zero — APPROVED** [review]
ACK, thanks for updating the PR and taking care of this.

**Pull request merged**